pkg/foxtrot/db.go

```diff
@@ -0,0 +1,70 @@
+package foxtrot
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+
+	"foxygo.at/s/errs"
+)
+
+var errDBInitialisation = errors.New("database initialisation error")
+
+type db struct {
+	conn *sql.DB
+}
+
+func newDB(dsnURI string) (*db, error) {
+	conn, err := sql.Open("sqlite3", dsnURI)
+	if err != nil {
+		return nil, errs.New(errDBInitialisation, err)
+	}
+
+	db := &db{conn: conn}
+	if err := db.setupSchema(); err != nil {
+		_ = db.conn.Close()
```

Comment:
i'd use `db.close()` here i think

Reply:
yes! I initially had prepared statements which I closed in db.close, which caused a panic (even with not nil guard).....
this was a temporary fix. 
done.